modules/resource_group_storage/variables.tf

```
@@ -0,0 +1,57 @@
+variable "resource_group_name" {
+  description = "The name of the resource group"
+  type        = string
+}
+
+variable "location" {
+  description = "The Azure region where resources will be created"
+  type        = string
+  default     = "West Europe"
+}
+
+variable "storage_account_name" {
+  description = "The name of the storage account"
+  type        = string
+}
```

The module accepts `storage_account_name` but does not validate it. Azure storage account names must be lowercase alphanumeric and 3–24 characters long. Consider adding a `validation` block to fail fast with a clear message, e.g. require a regex `^[a-z0-9]{3,24}$` so invalid names are caught before provider apply. This will prevent confusing runtime errors.